app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt

```diff
@@ -1416,4 +1421,177 @@ class PatientRepositoryAndroidTest {
     assertThat(patientRepository.findPatientWithBusinessId(deletedBusinessIdentifier).blockingFirst()).isEqualTo(None)
     assertThat(patientRepository.findPatientWithBusinessId("missing_identifier").blockingFirst()).isEqualTo(None)
   }
+  
+  @Test
+  fun if_patient_is_defaulter_then_isPatientDefaulter_should_return_true() {
+
+    val defaulterPatients = arrayListOf<String>()
+
+    fun savePatientRecord(
+        fullName: String,
+        bpMeasurement: List<BloodPressureMeasurement>?,
+        hasHadHeartAttack: MedicalHistory.Answer = NO,
+        hasHadStroke: MedicalHistory.Answer = NO,
+        hasDiabetes: MedicalHistory.Answer = NO,
+        hasHadKidneyDisease: MedicalHistory.Answer = NO,
+        protocolDrug: ProtocolDrug?,
+        appointmentDate: LocalDate?
+    ) {
+      val patientUuid = patientRepository.saveOngoingEntry(testData.ongoingPatientEntry(fullName = fullName))
+          .andThen(patientRepository.saveOngoingEntryAsPatient())
+          .blockingGet()
+          .uuid
+
+      bpMeasurement?.forEach {
+        bloodPressureRepository.saveMeasurement(patientUuid, it.systolic, it.diastolic, it.createdAt).blockingGet()
+      }
+      medicalHistoryRepository.save(listOf(testData.medicalHistory(
+          patientUuid = patientUuid,
+          hasDiabetes = hasDiabetes,
+          hasHadHeartAttack = hasHadHeartAttack,
+          hasHadKidneyDisease = hasHadKidneyDisease,
+          hasHadStroke = hasHadStroke))).blockingAwait()
+
+      protocolDrug?.let {
+        prescriptionRepository.savePrescription(patientUuid = patientUuid, drug = it).blockingAwait()
+      }
+
+      appointmentDate?.let {
+        appointmentRepository.schedule(patientUuid = patientUuid, appointmentDate = it).blockingGet()
+      }
+
+      val isDefaulter = patientRepository.isPatientDefaulter(patientUuid).blockingFirst()
```

Comment:
This is a side effect, and should not be done here. What should be done instead is that this method should return the generated `uuid`, and we should retain them and run the assertions after saving all the patients in the database.

Reply:
I agree that I should not combine the two operations.
Let me try to refactor this.

Reply:
Refactored this.